modules/kmm-building-and-signing-a-moduleloader-container-image.adoc

```diff
@@ -0,0 +1,81 @@
+// Module included in the following assemblies:
+//
+// * hardware_enablement/kmm-kernel-module-management.adoc
+
+:_content-type: PROCEDURE
+[id="kmm-building-and-signing-a-moduleloader-container-image_{context}"]
+= Building and signing a ModuleLoader container image
+
+Use this procedure if you have source code and need to build your image first.
+
+The YAML file shown below should build a new container image using the source code from the repository. The image produced is saved back in the registry with a temporary name, and this temporary image is
+then signed using the parameters in the `sign` section.
+
+The temporary image name is based on the final image name and is set to be `<containerImage>:<tag>-<namespace>_<module name>_kmm_unsigned`.
+
+For example, using the YAML below Kernel Module Management (KMM) builds an image named `quay.io/chrisp262/minimal-driver:final-default_example-module_kmm_unsigned` containing the build with unsigned kmods and push it to the registry. Then it creates a second image named `quay.io/chrisp262/minimal-driver:final` that contains the signed kmods.
+It is this second image that is loaded by the DaemonSet and deploys the kmods to the cluster nodes.
+
+Once it is signed, the temporary image can be safely deleted from the registry. It will be rebuilt, if needed.
+
+.Prerequisites
+
+* The `keySecret` and `certSecret` secrets have been created.
+
+.Procedure
+
+. Apply the YAML file:
++
+[source,yaml]
+----
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: example-module-dockerfile
+  namespace: default
+data:
+  Dockerfile: |
+    ARG DTK_AUTO
+    ARG KERNEL_VERSION
+    FROM ${DTK_AUTO} as builder
+    WORKDIR /build/
+    RUN git clone -b main --single-branch https://github.com/rh-ecosystem-edge/kernel-module-management.git
+    WORKDIR kernel-module-management/ci/kmm-kmod/
+    RUN make
+    FROM docker.io/redhat/ubi8:latest
```

Comment:
Maybe it will look better to use the official Red Hat registry ( registry.access.redhat.com/ubi8/ubi:latest ) than using the docker.io. WDYT ?

Reply:
Made this change.